diff --git a/plugins/vpp/binapi/vpp2001/fib_types/fib_types.ba.go b/plugins/vpp/binapi/vpp2001/fib_types/fib_types.ba.go
index 3bdb5510a..191e53c00 100644
--- a/plugins/vpp/binapi/vpp2001/fib_types/fib_types.ba.go
+++ b/plugins/vpp/binapi/vpp2001/fib_types/fib_types.ba.go
@@ -458,8 +458,8 @@ type FibPath struct {
	Flags      FibPathFlags
	Proto      FibPathNhProto
	Nh         FibPathNh
-	NLabels    uint8
-	LabelStack []FibMplsLabel `struc:"[16]FibMplsLabel"`
+	NLabels    uint8 `struc:"sizeof=LabelStack"` // MANUALLY FIXED, see https://jira.fd.io/browse/VPP-1261
+	LabelStack []FibMplsLabel
 }

 func (*FibPath) GetTypeName() string {
