diff --git a/plugins/vpp/binapi/vpp2001_324/sr/sr.ba.go b/plugins/vpp/binapi/vpp2001_324/sr/sr.ba.go
index 775dac6f4..c2847d341 100644
--- a/plugins/vpp/binapi/vpp2001_324/sr/sr.ba.go
+++ b/plugins/vpp/binapi/vpp2001_324/sr/sr.ba.go
@@ -49,9 +49,9 @@ func (*Srv6Sid) GetTypeName() string {

 // Srv6SidList represents VPP binary API type 'srv6_sid_list'.
 type Srv6SidList struct {
-	NumSids uint8
+	NumSids uint8 `struc:"sizeof=Sids"`
	Weight  uint32
-	Sids    []Srv6Sid `struc:"[16]Srv6Sid"`
+	Sids    []Srv6Sid
 }

 func (*Srv6SidList) GetTypeName() string {
