diff --git b/plugins/vpp/binapi/abf/abf.ba.go a/plugins/vpp/binapi/vpp1810/abf/abf.ba.go
index 5f2ab43b1..9efa3beee 100644
--- a/plugins/vpp/binapi/default/abf/abf.ba.go
+++ b/plugins/vpp/binapi/default/abf/abf.ba.go
@@ -96,8 +96,8 @@ type FibPath struct {
 	NextHopID         uint32
 	RpfID             uint32
 	ViaLabel          uint32
-	NLabels           uint8
-	LabelStack        []FibMplsLabel `struc:"[16]FibMplsLabel"`
+	NLabels           uint8 `struc:"sizeof=LabelStack"` // MANUALLY FIXED
+	LabelStack        []FibMplsLabel
 }
 
 func (*FibPath) GetTypeName() string {
