diff --git b/plugins/vpp/binapi/ip/ip.go a/plugins/vpp/binapi/ip/ip.go
index 1bcfcef9..a14d15b5 100644
--- b/plugins/vpp/binapi/ip/ip.go
+++ a/plugins/vpp/binapi/ip/ip.go
@@ -152,8 +152,8 @@
 	NextHopID         uint32
 	RpfID             uint32
 	ViaLabel          uint32
-	NLabels           uint8
-	LabelStack        []FibMplsLabel `struc:"[16]FibMplsLabel"`
+	NLabels           uint8 `struc:"sizeof=LabelStack"` // MANUALLY FIXED, see https://jira.fd.io/browse/VPP-1261
+	LabelStack        []FibMplsLabel
 }
 
 func (*FibPath) GetTypeName() string {
